src/main/java/org/reaktivity/nukleus/http2/internal/stream/Http2ServerFactory.java

```diff
@@ -2673,51 +2742,59 @@ private void doRequestBegin(
                 onResponseWindowUpdate(traceId, authorization, remoteSettings.initialWindowSize);
             }
 
-            private void doRequestData(
+            private int doRequestData(
                 long traceId,
                 long authorization,
                 DirectBuffer buffer,
                 int offset,
                 int limit)
             {
+                int progress = 0;
                 assert Http2State.initialOpening(state);
 
-                final int length = limit - offset;
-
-                localBudget -= length;
+                final int maxLength = limit - offset;
+                localBudget -= maxLength;
 
                 if (localBudget < 0)
                 {
                     doEncodeRstStream(traceId, authorization, streamId, Http2ErrorCode.FLOW_CONTROL_ERROR);
                     cleanup(traceId, authorization);
+                    progress += maxLength;
```

Comment:
Is this actually necessary? If feels like it impacts the network side decode which is far removed from this part of the code.